docs/application/dotnet/guides/nui/visuals.md

```diff
@@ -1,390 +1,552 @@
 # Visuals
 
-Visuals are the main building block for UI components. They provide reusable rendering logic that is controlled using properties and can be used by all components. They also respond to view size and color changes, and can perform clipping at the renderer level.
+Visuals are the main building blocks for UI components. They provide reusable rendering logic that is controlled by using properties and can be used by all components. They also respond to view size and color changes, and can perform clipping at the renderer level.
 
-Every UI component has three states: `NORMAL`, `FOCUSED`, and `DISABLED`. In addition, buttons have two substates for each state: `SELECTED` and `UNSELECTED`. A visual must be defined for each state and substate. You can share the same visual across multiple states. The components's current state or substate determines which visuals are shown. For more information on styling and transitioning between state visuals, see [Styling Components with JSON](styling-controls-with-JSON.md).
 
-NUI provides the visual types, such as the following:
+Properties specific for each individual Visual are encapsulated in the corresponding class: a 'Border' Visual is defined in a BorderVisual class, a 'Color' Visual in a ColorVisual class, and so on. 
+Properties that are common for all Visual types are inherited from a [Tizen.NUI.VisualMap](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.VisualMap.html) class.
+To render a Visual it has to be added to a view. A container class that controls if a user adds any Visual to it is a [Tizen.NUI.BaseComponents.VisualView](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.BaseComponents.VisualView.html).
 
--   [Border](#bordervisual)
--   [Color](#colorvisual)
--   [Gradient](#gradientvisual)
--   [Image](#imagevisual)
--   [Mesh](#meshvisual)
--   [Primitive](#primitivevisual)
--   [Text](#textvisual)
--   [Wireframe](#wireframevisual)
+In the following sections on this guide one can find:
```

Comment:
and what would you say for the following:
Follow the steps in this section to familiarize yourself with visuals:

Reply:
@mgrudzinska 

Since we are on the same page, and when we say "Follow the steps to help yourself" , it is implicit that steps are following up. 

Both forms are correct.

Reply:
I removed the agenda after the comment in the 'Pagination' PR #1366 
please let me know whether are you ok with that